sql/sql_cache.cc

```diff
@@ -991,14 +991,14 @@ void Query_cache_query::unlock_n_destroy()
 
 extern "C"
 {
-uchar *query_cache_query_get_key(const uchar *record, size_t *length,
-				my_bool not_used)
-{
-  Query_cache_block *query_block = (Query_cache_block*) record;
-  *length = (query_block->used - query_block->headers_len() -
-	     ALIGN_SIZE(sizeof(Query_cache_query)));
-  return (((uchar *) query_block->data()) +
-	  ALIGN_SIZE(sizeof(Query_cache_query)));
+  const uchar *query_cache_query_get_key(const void *record, size_t *length,
+                                         my_bool)
+  {
+    auto query_block= static_cast<const Query_cache_block *>(record);
+    *length= (query_block->used - query_block->headers_len() -
+              ALIGN_SIZE(sizeof(Query_cache_query)));
+    return reinterpret_cast<const uchar *>(
+        ((query_block->data()) + ALIGN_SIZE(sizeof(Query_cache_query))));
```

Comment:
I’d split the line one character earlier, between the `>` and `(` to remain compatible with the default settings of GNU Emacs `cc-mode`.